src/Simplex/FileTransfer/Agent.hs

```diff
@@ -194,6 +194,7 @@ runXFTPRcvWorker c srv Worker {doWork} = do
         (fc@RcvFileChunk {userId, rcvFileId, rcvFileEntityId, digest, fileTmpPath, replicas = replica@RcvFileChunkReplica {rcvChunkReplicaId, server, delay} : _}, approvedRelays) -> do
           let ri' = maybe ri (\d -> ri {initialInterval = d, increaseAfter = 0}) delay
           withRetryIntervalLimit xftpConsecutiveRetries ri' $ \delay' loop -> do
+            atomically $ waitWhileSuspended c
```

Comment:
seems somewhat redundant with assertAgentForeground check (line 187 - they check the same state slightly differently), maybe only this check is required? same for other workers?

Reply:
line 187 - outside of the loop
this is inside retry loop - without it it may enter the loop and never suspend